dsl/parse_test.go

```diff
@@ -99,6 +102,49 @@ func TestValidParseResponseStep(t *testing.T) {
 
 	// Ensure the slugs align with what we want
 	assert.Equal(t, hop.Ons[0].Calls[0].Slug, "a_sensor-first_task")
+
+	// Ensure the done block is empty
+	assert.Nil(t, hop.Ons[0].Done)
+}
+
+func TestValidParseDone(t *testing.T) {
+	logger := logs.NoOpLogger()
+	ctx := context.Background()
+
+	hopsFile := "./testdata/valid.hops"
```

Comment:
I'd prefer to see the relevant hops config and event here, otherwise it's quite tricky to see what is being tested. May not be possible, I understand.

Reply:
I'm with you, but personally I feel like using large fixtures from files is an accepted standard and for this kind of test perferrable.

When we're testing smaller pieces of pipelines I tend to inline them, but I'd hate to see this level of data as a string embedded in go code.